### DISCUSSION TITLE
Nested User / Role variables in Conditions?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#15695] [Archive]
Currently, as said [here](https://docs.directus.io/reference/filter-rules.html#dynamic-variables) we have
>**Nested User / Role variables in Permissions**
When configuring permissions, $CURRENT_USER and $CURRENT_ROLE allow you to specify 
**any (nested) field under the current user/role** 
as well as the root ID. For example: $CURRENT_ROLE.name or $CURRENT_USER.avatar.filesize.
**This includes custom fields that were added to the directus_users/directus_roles tables.**
_Note: This feature is only available for permissions, validation, and presets. 
Regular filters and conditional fields currently only support the root ID._

I'd like to know what prevents getting (nested) field under the current user/role in conditions?

Without this feature, I'll have to multiply the roles tens times and face a heavy roles management.

**Comment by br41nslug** [comment]
Currently the conditions/filters are using the `accountability` object which only contains the UUIDs. For this to be supported this logic would have to fetch any required nested data from the API which could cause a noticeable performance lag because this conditional logic can run on any keypress in the form to check if a condition has been met 😬

**Reply by Pachat** [reply]
Thank you for your explanation. I understand that performance must be preserved.

My use case is about 10 roles spread though 30 groups of users. A few fields of the collection are private to a group.

So, the solution I see, is to have 10 * 30 roles, which is rather tough to manage.

> For instance, to add a field common to the 30 groups, I'll have to set 60 permissions (Read and Write ) of this field.

I do not find a better way that fits Directus requirements.

**Reply by br41nslug** [reply]
perhaps this suggestion could help https://github.com/directus/directus/discussions/15347

**Reply by Pachat** [reply]
Yes. that's a very good suggestion that precisely joins what i am implementing now this way:

- add a custom field to the `directus_roles` as `parent_role`
- use a bespoke DB (PostgreSQL) trigger directly on `directus_roles.UPDATE` that acts on `directus_permissions` to handle inheritance.
